lib/user_space/uthread.c

```diff
@@ -130,15 +130,22 @@ M0_INTERNAL int m0_thread_confine(struct m0_thread *q,
 				  const struct m0_bitmap *processors)
 {
 	size_t    idx;
+	size_t    idx1 = M0_PROCESSORS_INVALID_ID;
 	size_t    nr_bits = min64u(processors->b_nr, CPU_SETSIZE);
 	cpu_set_t cpuset;
 
+	M0_PRE(m0_bitmap_set_nr(processors) > 0);
+
 	CPU_ZERO(&cpuset);
 
 	for (idx = 0; idx < nr_bits; ++idx) {
-		if (m0_bitmap_get(processors, idx))
+		if (m0_bitmap_get(processors, idx)) {
 			CPU_SET(idx, &cpuset);
+			if (idx1 == M0_PROCESSORS_INVALID_ID)
+				idx1 = idx;
```

Comment:
From the code, I can conclude that:
idx1 is the first CPUID that this thread can run on.
Am I correct?

Reply:
Yes.